witchcraft-server/src/lib.rs

```diff
@@ -354,10 +354,26 @@ where
     I: AsRef<InstallConfig> + DeserializeOwned,
     R: AsRef<RuntimeConfig> + DeserializeOwned + PartialEq + 'static + Sync + Send,
     F: FnOnce(I, Refreshable<R, Error>, &mut Witchcraft) -> Result<(), Error>,
+{
+    init_with_configs(init, configs::load_install::<I>, configs::load_runtime::<R>)
+}
+
+/// Initializes a Witchcraft server with custom config loaders.
+///
+/// `init` is invoked with the install and runtime configs from the provided loaders as well as the [`Witchcraft`]
+/// context object. It is expected to return quickly; any long running initialization should be spawned off into
+/// the background to run asynchronously.
+pub fn init_with_configs<I, R, F, LI, LR>(init: F, load_install: LI, load_runtime: LR)
```

Comment:
Do you need to defer the computation here, or can we just have this take `install: I, runtime: Refreshable<R, Error>` instead?

Reply:
Hm my thinking was since loading can fail (both for custom and the default loaders) we want to defer the actual load post logging initialization to make sure the details are captured.

Reply:
I think it's reasonable to say that the caller should be responsible for handling and logging for their own errors if they want to use this entry point, but I think we should still defer loading the configs in the default code path since it's fairly common to have deserialization issues which we'd want to be logged properly.

Reply:
Oh sure that makes sense.